contracts/contracts/coordination/subscription/Subscription.sol

```diff
@@ -0,0 +1,279 @@
+
+    // Mapping from (ritualId, address) to subscription ID
+    mapping(bytes32 => uint32) public subscribers;
+
+    uint32 public numberOfSubscriptions;
+
+    // TODO: DAO Treasury
+    // TODO: Should it be updatable?
+    address public immutable beneficiary;
+
+    /**
+     * @notice Emitted when a subscription is created
+     * @param subscriptionId The ID of the subscription
+     * @param subscriber The address of the subscriber
+     * @param ritualId The ID of the ritual
+     */
+    event SubscriptionCreated(
+        uint32 indexed subscriptionId,
+        address indexed subscriber,
+        uint32 indexed ritualId
+    );
+
+    /**
+     * @notice Emitted when a subscription is cancelled
+     * @param subscriptionId The ID of the subscription
+     * @param subscriber The address of the subscriber
+     * @param ritualId The ID of the ritual
+     */
+    event SubscriptionCancelled(
+        uint32 indexed subscriptionId,
+        address indexed subscriber,
+        uint32 indexed ritualId
+    );
+
+    /**
+     * @notice Emitted when a subscription is paid
+     * @param subscriptionId The ID of the subscription
+     * @param subscriber The address of the subscriber
+     * @param amount The amount paid
+     */
+    event SubscriptionPaid(
+        uint32 indexed subscriptionId,
+        address indexed subscriber,
+        uint256 amount
+    );
+
+    /**
+     * @notice Emitted when a subscription is spent
+     * @param beneficiary The address of the beneficiary
+     * @param amount The amount withdrawn
+     * @param amount The amount spent
+     */
+    event WithdrawalToBeneficiary(address indexed beneficiary, uint256 amount);
+
+    /**
+     * @notice Sets the coordinator and fee token contracts
+     * @dev The coordinator and fee token contracts cannot be zero addresses
+     * @param _coordinator The address of the coordinator contract
+     * @param _feeToken The address of the fee token contract
+     * @param _beneficiary The address of the beneficiary
+     */
+    constructor(Coordinator _coordinator, IERC20 _feeToken, address _beneficiary) {
+        require(address(_coordinator) != address(0), "Coordinator cannot be the zero address");
+        require(address(_feeToken) != address(0), "Fee token cannot be the zero address");
+        require(_beneficiary != address(0), "Beneficiary cannot be the zero address");
+        coordinator = _coordinator;
+        feeToken = _feeToken;
+        beneficiary = _beneficiary;
+    }
+
+    /**
+     * @notice Returns the subscription fee
+     * @return The subscription fee
+     */
+    function subscriptionFee() public pure returns (uint256) {
+        return 42 * 10 ** 20; // TODO
+    }
+
+    /**
+     * @notice Returns the base expiration duration
+     * @return The base expiration duration
+     */
+    function baseExpiration() public pure returns (uint256) {
+        return 52 weeks; // TODO
```

Comment:
Do these "TODOs" still need to be revisited?

Reply:
I plan to remove Subscription contract when ManagedAllowList will be integrated with new Subscription